app/controllers/TransactionalController.scala

```diff
@@ -84,5 +84,28 @@ trait TransactionalController extends BaseController {
         case _ => NoContent
       }
   }
+
+  def fetchSicCodesByTransactionID(txId: String) = Action.async {
```

Comment:
this logic should reall be in a service as the only decision  of whether to return 200 or 204 is whether there is Some(SicCodes).
service.fetchSicCodes(txId, usePublicAPI = false could have been assigned to a scoped lazy val as it called twice with the same params
- Also why in this function are we
lazy val txIdSicCodes = service.fetchSicCodes(txId, usePublicAPI = false)